### ISSUE TITLE
No $var expansion in segment URL generation

### ISSUE THREAD

**Issue by bson** [kaltura/nginx-vod-module#334]
We're experimenting with various services in front of our main CDN, so requests can come in through different CNAMEs and even proxy services depending on what URL we generate.  So I configured:

            vod_segments_base_url http://$http_host;

When I test this on a variant m3u8 it works as expected:

> $ curl -H "Host: ts.example.com" http://localhost/path/xhogua0dnhl4r37_2683601.mp4/master.m3u8
> #EXTM3U
> #EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=675280,RESOLUTION=640x360,CODECS="avc1.4d401e,mp4a.40.2"
> http://ts.example.com/path/xhogua0dnhl4r37_2683601.mp4/index-v1-a1.m3u8

However, when fetching the media playlist, I get:

> $ curl -s -H "Host: ts.example.com" http://localhost/path/xhogua0dnhl4r37_2683601.mp4/index-v1-a1.m3u8 | sed 20q
> #EXTM3U
> #EXT-X-TARGETDURATION:10
> #EXT-X-ALLOW-CACHE:YES
> #EXT-X-PLAYLIST-TYPE:VOD
> #EXT-X-KEY:METHOD=AES128,URI="http://ts.example.com/path/xhogua0dnhl4r37_2683601.mp4/encryption.key"
> #EXT-X-VERSION:3
> #EXT-X-MEDIA-SEQUENCE:1
> #EXTINF:7.007,
> http://$http_host/path/xhogua0dnhl4r37_2683601.mp4/seg-1-v1-a1.ts
> #EXTINF:0.000,
> http://$http_host/path/xhogua0dnhl4r37_2683601.mp4/seg-2-v1-a1.ts

It would be very useful to perform $var expansion in the TS segment URLs as well...
Note that the encryption key URL gets the proper TLC.

**Comment by erankor** [comment]
I believe you are not using the latest master version, since this feature was added about a month ago (it is possible that our pre-compiled package is a bit behind btw).
The reason it works for you on the master playlist is that `vod_segments_base_url` affects only the media playlist, it has no effect on the master playlist, and the default behavior of the module is to use the incoming host header.
So, to put it differently, there is probably no good reason to do `vod_segments_base_url http://$http_host;`, just omit this config var and it should give you what you want. 
Or, alternatively, you can upgrade to the latest version, and it should work as well.

**Comment by bson** [comment]
Oh, interesting.  When I originally evaluated the module it kept trying to use what nginx determined was the host's fqdn.  Seems this has indeed been changed since - excellent!

We haven't updated yet to the latest commit(s)... it's not so easy for us to just make changes to the segmentation since it requires invalidating our CDN caches.

**Comment by bson** [comment]
Not a problem.  Resolved.
